src/plugins/validation/2and3/semantic-validators/refs.js

```diff
@@ -29,21 +25,52 @@ module.exports.validate = function({ jsSpec, specStr, isOAS3 }) {
     match = refRegex.exec(specStr);
   }
 
+  const definitionSectionName = isOAS3 ? 'components' : 'definitions';
   // de-dupe the array, and filter out non-definition refs
   const definitionsRefs = filter(uniq(refs), v =>
-    startsWith(v, `#/${basePath.join('/')}`)
+    startsWith(v, `#/${definitionSectionName}`)
   );
 
-  const definitions = isOAS3 ? jsSpec.components.schemas : jsSpec.definitions;
-  each(definitions, (def, defName) => {
-    if (definitionsRefs.indexOf(`#/${basePath.join('/')}/${defName}`) === -1) {
+  // checks if the definitions are used, and if not, record a warning
+  if (isOAS3) {
+    const definitionTypeList = ['schemas', 'parameters', 'responses'];
```

Comment:
I like the way you're doing this. Since it's so simple to check the different locations - we might as well look at all of them! Here is the [full list](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#components-object)

Reply:
I've seen that examples are often excluded from validation. walk.js, for example, does not walk down keys named 'example' or 'examples.'

Should I include or exclude examples from the full list?

Reply:
That's a good question. I think still include it. No, we can't validate examples, but we can still let a user know if they've defined one they aren't using